src/image/ImageManagerProtocol.cc

```diff
@@ -748,6 +845,142 @@ void ImageManager::_snap_flatten(unique_ptr<image_msg_t> msg)
 
 /* -------------------------------------------------------------------------- */
 
+void ImageManager::_restore(unique_ptr<image_msg_t> msg)
+{
+    NebulaLog::dddebug("ImM", "_restore: " + msg->payload());
+
+    if (msg->status() == "SUCCESS")
+    {
+        if (msg->payload().empty())
+        {
+            notify_request(msg->oid(), false,
+                           "Cannot get info about restored disk images");
+            return;
+        }
+
+        NebulaLog::log("ImM", Log::INFO, "Backup successfully restored: "
+                       + msg->payload());
+
+        notify_request(msg->oid(), true, msg->payload());
+    }
+    else
+    {
+        notify_request(msg->oid(), false, msg->payload());
+    }
+}
+
+/* -------------------------------------------------------------------------- */
+
+void ImageManager::_increment_flatten(unique_ptr<image_msg_t> msg)
+{
+    NebulaLog::dddebug("ImM", "_increment_flatten: " + msg->payload());
+
+    auto image = ipool->get(msg->oid());
+
+    if ( !image )
+    {
+        return;
+    }
+
+    bool error = false;
+
+    long long size_orig;
+    long long size_new;
+
+    int ds_id = image->get_ds_id();
+    int uid   = image->get_uid();
+    int gid   = image->get_gid();
+
+    auto ids   = image->get_running_ids();
+    auto first = ids.cbegin();
+
+    int vm_id  = -1;
+
+    if (first != ids.cend())
+    {
+        vm_id = *first;
+    }
+
+    if (msg->status() == "SUCCESS")
+    {
+        auto& increments = image->increments();
+
+        size_orig = image->get_size();
+
+        string size_mb;
+        string chain;
+
+        istringstream is(msg->payload());
+
+        is >> size_mb;
+
+        is >> chain;
+
+        int rc = increments.update_increments(chain, size_mb);
+
+        size_new = increments.total_size();
+
+        image->set_size(size_new);
+
+        error = rc == -1;
+    }
+    else
+    {
+        error = true;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

We observed that the presence of non-empty statements like `a + b` without clear context or purpose within the codebase can lead to potential confusion or misinterpretation. To mitigate this issue, it&#39;s advisable to ensure that such statements are appropriately integrated into larger constructs that contribute meaningfully to the program&#39;s logic or functionality. This could involve assigning the result of the expression to a variable.